projects/packages/my-jetpack/_inc/components/stats-section/index.jsx

```diff
@@ -1,16 +1,64 @@
+import { __ } from '@wordpress/i18n';
+import React, { useCallback } from 'react';
+import useAnalytics from '../../hooks/use-analytics';
+import { useProduct } from '../../hooks/use-product';
 import useStatsCounts from '../../hooks/use-stats-counts';
+import ProductCard from '../connected-product-card';
+import { PRODUCT_STATUSES } from '../product-card/action-button';
 import StatsCards from './cards';
-/**
- * StatsSection component that renders the Stats cards, passing down the stats counts from the store.
- *
- * @returns {object} The StatsSection component.
- */
+
 const StatsSection = () => {
+	const { detail } = useProduct( 'stats' );
+	const { status } = detail;
+	const { userIsAdmin } = window?.myJetpackInitialState ?? false;
 	const { statsCounts } = useStatsCounts();
 	const counts = statsCounts?.past_seven_days || {};
 	const previousCounts = statsCounts?.between_past_eight_and_fifteen_days || {};
+	const { recordEvent } = useAnalytics();
+
+	/**
+	 * Called when "See detailed stats" button is clicked.
+	 */
+	const onDetailedStatsClick = useCallback( () => {
+		recordEvent( 'jetpack_myjetpack_stats_card_seedetailedstats_click', {
+			product: 'stats',
+		} );
+	}, [ recordEvent ] );
+
+	const shouldShowSecondaryButton = useCallback(
+		() => !! ( status === PRODUCT_STATUSES.CAN_UPGRADE ),
+		[ status ]
+	);
+
+	const viewStatsButton = {
+		href: 'admin.php?page=stats',
+		label: __( 'View detailed stats', 'jetpack-my-jetpack' ),
+		onClick: onDetailedStatsClick,
+		shouldShowButton: shouldShowSecondaryButton,
+	};
+
+	// Override the primary action button to read "View detailed stats" instead
+	// of the default text, "View".
+	const primaryActionOverride = {
+		[ PRODUCT_STATUSES.ACTIVE ]: {
+			label: __( 'View detailed stats', 'jetpack-my-jetpack' ),
+		},
+		[ PRODUCT_STATUSES.ERROR ]: {
+			label: __( 'Connect Jetpack to use Stats', 'jetpack-my-jetpack' ),
+		},
+	};
 
-	return <StatsCards counts={ counts } previousCounts={ previousCounts } />;
+	return (
+		<ProductCard
+			admin={ userIsAdmin }
+			slug={ 'stats' }
```

Comment:
Very small nitpick, I see this is used here and in the tracks event call, could be slightly better to move this to a top level variable